mmv1/third_party/terraform/resources/resource_storage_bucket.go.erb

```diff
@@ -1119,13 +1125,19 @@ func expandBucketWebsite(v interface{}) *storage.BucketWebsite {
 }
 
 func expandIamConfiguration(d *schema.ResourceData) *storage.BucketIamConfiguration {
-	return &storage.BucketIamConfiguration{
+	cfg := &storage.BucketIamConfiguration{
 		ForceSendFields: []string{"UniformBucketLevelAccess"},
 		UniformBucketLevelAccess: &storage.BucketIamConfigurationUniformBucketLevelAccess{
 			Enabled:         d.Get("uniform_bucket_level_access").(bool),
 			ForceSendFields: []string{"Enabled"},
 		},
 	}
+<% unless version == "ga" -%>
+	if v, ok := d.GetOk("public_access_prevention"); ok {
+		cfg.PublicAccessPrevention = v.(string)
+	}
+<% end -%>
+	return cfg
```

Comment:
```suggestion
	cfg := &storage.BucketIamConfiguration{}
	if v, ok := d.GetOk("uniform_bucket_level_access"); ok {
		cfg.ForceSendFields = []string{"UniformBucketLevelAccess"},
		cfg.UniformBucketLevelAccess = &storage.BucketIamConfigurationUniformBucketLevelAccess{
				  Enabled:         d.Get("uniform_bucket_level_access").(bool),
				  ForceSendFields: []string{"Enabled"},
			  }
	if v, ok := d.GetOk("public_access_prevention"); ok {
		cfg.PublicAccessPrevention = v.(string)
	}
	return cfg
```

This is so we do not change existing behavior

Reply:
Happy to make this change, but this is an alteration of the current code, no? `uniform_bucket_level_access` was already being set when I made these changes, all I've done ias set `public_access_prevention`.

Reply:
ah this was a misunderstanding on my part... you are correct ~